mm2src/coins/z_coin/storage/walletdb/mod.rs

```diff
@@ -33,27 +48,74 @@ pub struct WalletDbShared {
     ticker: String,
 }
 
+#[cfg(not(target_arch = "wasm32"))]
+async fn checkpoint_block_from_height(
```

Comment:
I think this entire fn should be in `z_rpc` module instead. It can be even a method of `ZRpcOps` trait with default implementation.

Reply:
@shamardy suggested the placement here for reuse across multiple UTXO enabled coins

Reply:
done